setup.py

```diff
@@ -66,11 +66,11 @@ def read(filename):
     ],
     extras_require={
         "notebook": [
-            # Notebook upper limit is a workaround for issues with IPython not being defined.
-            "notebook>=4.4.1,<7",
+            "notebook>=7",
             "ipywidgets>=7.0.0",
-            "jupyter_client<8",  # https://github.com/jupyter/notebook/issues/6748
-            "pyzmq<25",  # https://github.com/jupyter/notebook/issues/6748
+            "jupyter_client>=8",
+            "ipympl",
```

Comment:
It would be nice to have some minimum defined here.

Reply:
Yeah, good catch on this front actually, since we need the latest `ipympl` for the version of `matplotlib` that we are using.